From 28818e6bd58512530b339a1a9232af82c252c7c9 Mon Sep 17 00:00:00 2001
From: Davide Beatrici <git@davidebeatrici.dev>
Date: Fri, 7 Jun 2024 20:12:58 +0200
Subject: [PATCH 2/2] winepulse: Don't set a floor for the period(s).

This driver, unlike the others, queries the engine for the actual device period.

The default period floor is not needed anymore because it now lives in mmdevapi.
The minimum period floor never mattered because exclusive mode is not supported.
---
 dlls/winepulse.drv/pulse.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git b/dlls/winepulse.drv/pulse.c a/dlls/winepulse.drv/pulse.c
index 068245f..dd8d0b4 100644
--- b/dlls/winepulse.drv/pulse.c
+++ a/dlls/winepulse.drv/pulse.c
@@ -113,6 +113,9 @@ static pa_mainloop *pulse_ml;
 static struct list g_phys_speakers = LIST_INIT(g_phys_speakers);
 static struct list g_phys_sources = LIST_INIT(g_phys_sources);
 
+static const REFERENCE_TIME MinimumPeriod = 30000;
+static const REFERENCE_TIME DefaultPeriod = 100000;
+
 static pthread_mutex_t pulse_mutex;
 static pthread_cond_t pulse_cond = PTHREAD_COND_INITIALIZER;
 
@@ -747,6 +750,12 @@ static void pulse_probe_settings(int render, const char *pulse_name, WAVEFORMATE
     if (length)
         *def_period = *min_period = pa_bytes_to_usec(10 * length, &ss);
 
+    if (*min_period < MinimumPeriod)
+        *min_period = MinimumPeriod;
+
+    if (*def_period < DefaultPeriod)
+        *def_period = DefaultPeriod;
+
     wfx->wFormatTag = WAVE_FORMAT_EXTENSIBLE;
     wfx->cbSize = sizeof(WAVEFORMATEXTENSIBLE) - sizeof(WAVEFORMATEX);
 
-- 
2.45.2

